kornia/augmentation/_2d/intensity/random_channel_dropout.py

```diff
@@ -0,0 +1,84 @@
+        fill_value (int, float): pixel value for the dropped channel.
+        same_on_batch: apply the same transformation across the batch.
+        p: probability of applying the transformation.
+        keepdim: whether to keep the output shape the same as input (True) or broadcast it
+                 to the batch form (False).
+    Shape:
+        - Input: :math:`(C, H, W)` or :math:`(B, C, H, W)`
+        - Output: :math:`(B, C, H, W)` or :math:`(C, H, W)`
+    Examples:
+        >>> _ =torch.random.manual_seed(1)
+        >>> img = torch.ones(1, 3, 5, 5)
+        >>> out = RandomChannelDropout(p=1)(img)
+        >>> out.shape
+        torch.Size([1, 3, 5, 5])
+        >>> out
+        tensor([[[[0., 0., 0., 0., 0.],
+                  [0., 0., 0., 0., 0.],
+                  [0., 0., 0., 0., 0.],
+                  [0., 0., 0., 0., 0.],
+                  [0., 0., 0., 0., 0.]],
+        <BLANKLINE>
+                 [[1., 1., 1., 1., 1.],
+                  [1., 1., 1., 1., 1.],
+                  [1., 1., 1., 1., 1.],
+                  [1., 1., 1., 1., 1.],
+                  [1., 1., 1., 1., 1.]],
+        <BLANKLINE>
+                 [[1., 1., 1., 1., 1.],
+                  [1., 1., 1., 1., 1.],
+                  [1., 1., 1., 1., 1.],
+                  [1., 1., 1., 1., 1.],
+                  [1., 1., 1., 1., 1.]]]])
+    """
+
+    def __init__(
+        self,
+        channel_drop_range: Tuple[int, int] = (1, 1),
+        fill_value: Union[int, float] = 0,
+        same_on_batch: bool = False,
+        p: float = 0.5,
+        keepdim: bool = False,
+    ) -> None:
+        super().__init__(p=p, same_on_batch=same_on_batch, p_batch=1.0, keepdim=keepdim)
+        self.fill_value = fill_value
+        KORNIA_CHECK(
+            isinstance(channel_drop_range, tuple) and len(channel_drop_range) == 2,
+            "Invalid channel_drop_range. Should be tuple of length 2.",
+        )
+        KORNIA_CHECK(
+            1 <= channel_drop_range[0] <= channel_drop_range[1],
+            "Invalid channel_drop_range. Max channel should be greater than lower.",
+        )
+        self.min_channel = channel_drop_range[0]
+        self.max_channel = channel_drop_range[1]
+
+    def apply_transform(
+        self, input: Tensor, params: Dict[str, Tensor], flags: Dict[str, Any], transform: Optional[Tensor] = None
+    ) -> Tensor:
+        KORNIA_CHECK(
+            (input.shape[-3] != 1 and len(input.shape) != 2), "Only one channel. ChannelDropout is not defined."
+        )
+        num_channels = input.shape[-3]
+        KORNIA_CHECK(self.max_channel < num_channels, "Can not drop all channels in ChannelDropout.")
+
+        num_channels_to_drop = torch.randint(low=self.min_channel, high=(self.max_channel + 1), size=(1,))[
+            0
+        ]  # +1 because highest integer to be drawn is not included
+
+        channels_to_drop = torch.randint(low=0, high=num_channels, size=(num_channels_to_drop,))
```

Comment:
i belive this should be from a random generator or be at least in the `generate_parameter` method like in https://github.com/kornia/kornia/blob/master/kornia/augmentation/_2d/intensity/motion_blur.py#L83

@shijianjian can point out how to do this more correctly
```suggestion
```